tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php

```diff
@@ -413,8 +413,6 @@ private static function assertAnalysis(AbstractControlCaseStructuresAnalysis $ex
         $serializeActual = serialize($analysis);
 
         if ($serializeExpected === $serializeActual) {
-            static::assertTrue(true);
```

Comment:
Now when I think about it again, shouldn't it be:

```php
static::markTestSkipped();
```

Reply:
no, because it would say that the test is not fully executed, while it is - if serialized form is same, no need to check all other assertions anymore.

but with just `return`, we had warning of PHPUnit like "test with no assertions, wtf?"

Reply:
Lacking assertion in any test would fail CI, which did not happen, this change is not making any test losing it's only assertion.

Reply:
I do not see value of this change and probably discussion about it is taking more time than value of any decision made here.

Here, we ___are___ doing the assertion.
We do [in pseudocode]
```
assertThat (
    [
        $serializeExpected === $serializeActual
    ]
    OR
    [
        static::assertSame($expectedAnalysis->getIndex(), $analysis->getIndex(), 'index');
        static::assertSame($expectedAnalysis->getOpenIndex(), $analysis->getOpenIndex(), 'open index');
        static::assertSame($expectedAnalysis->getCloseIndex(), $analysis->getCloseIndex(), 'close index');
        static::assertInstanceOf(\get_class($expectedAnalysis), $analysis);
        ...
    ]
)
```

Reply:
> Lacking assertion in any test would fail CI, which did not happen, this change is not making any test losing it's only assertion.

only because how the assertion is used __CURRENTLY__. The assertion should be working independently of the usecase. If you drop this assert bumping and having the custom assertion method to be the only one called in test method, you can see the test marked as risky. We should not have custom assertion that can blow-up in developer face, if they didn't use any other assertion in their test method.

```
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ git du
diff --git a/tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php b/tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php
index 6a3c26408..d10f601e6 100644
--- a/tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php
+++ b/tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php
@@ -41,7 +41,7 @@ public function testFindControlStructures(array $expectedAnalyses, string $sourc
         $tokens = Tokens::fromCode($source);
         $analyses = iterator_to_array(ControlCaseStructuresAnalyzer::findControlStructures($tokens, [T_SWITCH]));
 
-        static::assertCount(\count($expectedAnalyses), $analyses);
+        // static::assertCount(\count($expectedAnalyses), $analyses);
 
         foreach ($expectedAnalyses as $index => $expectedAnalysis) {
             self::assertAnalysis($expectedAnalysis, $analyses[$index]);
@@ -413,7 +413,7 @@ private static function assertAnalysis(AbstractControlCaseStructuresAnalysis $ex
         $serializeActual = serialize($analysis);
 
         if ($serializeExpected === $serializeActual) {
-            static::assertTrue(true);
+            // static::assertTrue(true);
 
             return;
         }
```

results in:
```
There was 1 risky test:

1) PhpCsFixer\Tests\Tokenizer\Analyzer\ControlCaseStructuresAnalyzerTest::testFindControlStructures with data set "two cases" (array(PhpCsFixer\Tokenizer\Analyzer\Analysis\SwitchAnalysis Object (...)), '<?php switch ($foo) {\n      ...     }')
This test did not perform any assertions

OK, but incomplete, skipped, or risky tests!
Tests: 26, Assertions: 5, Risky: 1.
```

Reply:
vide https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6874